main_app/urls.py

```diff
@@ -1,7 +1,8 @@
 from django.urls import path
 from django.views.generic import TemplateView
+from .views import MainPageView
 
 urlpatterns = [
-    path('', TemplateView.as_view(template_name='main_page.html'), name='main_page'),
+    path('', MainPageView.as_view(template_name='main_page.html'), name='main_page'),
```

Comment:
Remove `template_name` here. `template_name` in urls.py is fine when there's no logic in view at all